src/istgt_lu.c

```diff
@@ -1737,11 +1758,51 @@ istgt_lu_add_unit(ISTGT_Ptr istgt, CF_SECTION *sp)
 	ISTGT_TRACELOG(ISTGT_TRACE_DEBUG, "ConsistencyFactor %d\n",
 	    lu->consistency_factor);
 
+	if (lu->replication_factor > lu->desired_replication_factor) {
+		ISTGT_ERRLOG("Invalid config ReplicationFactor is"
+		    " greater than Desired ReplicationFactor\n");
+		goto error_return;
+	}
 	if (lu->replication_factor <= 0 || lu->consistency_factor <= 0 ||
 		lu->replication_factor < lu->consistency_factor) {
 		ISTGT_ERRLOG("Invalid ReplicationFactor/ConsistencyFactor or their ratio\n");
 		goto error_return;
 	}
+	/* Read known replica details from conf file and maintain in memory structure*/
+	val = istgt_get_val(sp, "Replica");
+	if (val == NULL) {
+		TAILQ_INIT(&lu->known_replica);
+	} else {
+		TAILQ_INIT(&lu->known_replica);
+		for (i=0; ; i++) {
+			key = istgt_get_nmval(sp, "Replica", i, 0);
+			if (key == NULL) {
+				break;
+			}
+			knr = xmalloc(sizeof (trusty_replica_t));
+			memset(knr, 0, sizeof(trusty_replica_t));
+			val = istgt_get_nmval(sp, "Replica", i, 1);
+			len = strlen(key);
+			knr->replica_id = (char *)malloc(sizeof(char)*len);
```

Comment:
you have to memset to 0

Reply:
allocate here one extra byte as well

Reply:
I will make sure that this kind of mistakes will not repeat.